assets/app/scripts/services/applicationGenerator.js

```diff
@@ -56,7 +56,7 @@ angular.module("openshiftConsole")
             {
               containerPort: parseInt(parts[0]), 
               name: input.name + "-" + parts[1] + "-" + parts[0],
-              protocol: parts[1]
```

Comment:
`image.dockerImageMetadata.Config.ExposedPorts` look like `8080/tcp`, so we'd need to handle those cases as well. It would also add uppercase TCP into the name as you mention.